apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/index.js

```diff
@@ -29,46 +40,53 @@ class PageTemplateModal extends Component {
 	state = {
 		isLoading: false,
 		previewedTemplate: null,
-		blocksByTemplateSlug: {},
-		titlesByTemplateSlug: {},
 		error: null,
 		isOpen: false,
 	};
 
-	constructor( props ) {
-		super();
-		const hasTemplates = ! isEmpty( props.templates );
-		this.state.isOpen = hasTemplates;
-		if ( hasTemplates ) {
-			// Select the first template automatically.
-			this.state.previewedTemplate = this.getDefaultSelectedTemplate( props );
-			// Extract titles for faster lookup.
-			this.state.titlesByTemplateSlug = mapValues( keyBy( props.templates, 'slug' ), 'title' );
-		}
-	}
-
-	componentDidMount() {
-		if ( this.state.isOpen ) {
-			trackView( this.props.segment.id, this.props.vertical.id );
-		}
+	// Extract titles for faster lookup.
+	getTitlesByTemplateSlugs = memoize( templates =>
+		mapValues( keyBy( templates, 'slug' ), 'title' )
+	);
 
-		// Parse templates blocks and store them into the state.
-		const blocksByTemplateSlug = reduce(
-			this.props.templates,
+	// Parse templates blocks and store them into the state.
```

Comment:
It doesn't seem we are saving to state anymore.

```suggestion
	// Parse templates blocks and memoize them.
```